File: v131016/TeeNew/Export_GIF.pas
unit Export_GIF;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeEngine, Series, GanttCh, TeeProcs, Chart, BubbleCh;

type
  TGIFExportForm = class(TBaseForm)
    Button1: TButton;
    Button2: TButton;
    SaveDialog1: TSaveDialog;
    Series1: TBubbleSeries;
    procedure Button1Click(Sender: TObject);
    procedure Button2Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

Uses TeeGIF, TeExport;

procedure TGIFExportForm.Button1Click(Sender: TObject);
begin
  TeeSavePanel(TGIFExportFormat,Chart1);
end;

procedure TGIFExportForm.Button2Click(Sender: TObject);
begin
  TeeExport(Self,Chart1);
end;

procedure TGIFExportForm.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(8);
end;

initialization
  RegisterClass(TGIFExportForm);
end.

File: TeeNew/Builder2006/Export_GIF.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Export_GIF.h"
#include "TeeGIF.hpp"
#include "TeExport.hpp"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeGIF"
#pragma resource "*.dfm"
TGIFExportForm *GIFExportForm;
//---------------------------------------------------------------------------
__fastcall TGIFExportForm::TGIFExportForm(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TGIFExportForm::Button1Click(TObject *Sender)
{
  TeeSavePanel(__classid(TGIFExportFormat),Chart1);
}
//---------------------------------------------------------------------------

void __fastcall TGIFExportForm::Button2Click(TObject *Sender)
{
  TeeExport(this,Chart1);
}
//---------------------------------------------------------------------------

void __fastcall TGIFExportForm::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(8);
}
//---------------------------------------------------------------------------

